### ISSUE TITLE
Support Symbol.iterator for ArrayProxy

### ISSUE THREAD

**Issue by dwickern** [emberjs/rfcs#243]
Using ember-data, you might write this code:
```
// blog-post.js
export default Model.extend({
  comments: hasMany('comment')
})

...

for (let comment in blogPost.get('comments')) {
  comment.deleteRecord()
}
```

But it would be wrong:
`TypeError: blogPost.get(...)[Symbol.iterator] is not a function`

Instead you have to call `blogPost.get('comments').toArray()`.

I would propose that ArrayProxy should also proxy `Symbol.iterator`. It can be implemented trivially:
```js
Ember.ArrayProxy.prototype[Symbol.iterator] = function() {
  return this.toArray()[Symbol.iterator]()
}
```

**Comment by buschtoens** [comment]
I'd rather add it in [`Ember.Enumerable`](https://www.emberjs.com/api/ember/2.14/classes/Ember.Enumerable), which [`Ember.ArrayProxy`](https://www.emberjs.com/api/ember/2.14/classes/Ember.ArrayProxy) uses.

I already brought this up in Slack once, but I think there were reservations, though I can't remember why exactly. I think it had something to with having to include the Babel polyfill.

**Comment by dwickern** [comment]
Good point, `Enumerable` might be a better place for it than `ArrayProxy`. In that case you would surely want a better `Symbol.iterator` implementation than my naive `toArray()` since you might be dealing with a lazy enumerable.

I can't see any harm in adding the method only when Symbol is available (native or polyfilled).

**Comment by maconfr** [comment]
Since ember v4.6 the [no-array-prototype-extensions](https://github.com/ember-cli/eslint-plugin-ember/blob/master/docs/rules/no-array-prototype-extensions.md) eslint rule has been activated.

The suggested patterns in the rule document don't apply to the `ArrayProxy`, because it is not iterable.
To avoid the lint errors, i need to replace the `.toArray()` Method by the `.content` Property or ignore the lint error. But this feels odd.
I think the `ArrayProxy` should be transparent so i don't have  to be aware of it until a really need the specific functionallity that this class provides.
